src/gnu/prolog/io/TermWriter.java

```diff
@@ -473,7 +473,14 @@ protected void displayAtom(WriteOptions options, AtomTerm atom)
 		if (options.quoted)
 		{
 			String value = atom.value;
-			print(needBeQuoted(value) ? getSingleQuoted(value) : value);
```

Comment:
The only reason I didn't was that in needBeQuoted we only have access to the underlying string, rather than the atom itself. I can't imagine anyone would ever want to redefine the string associated with the empty list atom, but a direct comparison with the reserved token seemed somehow 'safer'. I could always use TermConstants.emptyListAtom.value.equals(value) in needBeQuoted, if you prefer?